### ISSUE TITLE
Task Progress

### ISSUE THREAD

**Issue by ronbeltran** [samuelcolvin/arq#425]
In Python RQ there is this `job.meta` dictionary that can be used to set some custom task progress indication which is useful for showing in the UI, does ARQ have this?

https://python-rq.org/docs/jobs/

**Comment by vidjuheffex** [comment]
Yes same question, I was trying to use the underlying redis instance to just write to the job using `await redis.set(I HAVE NO CLUE WHAT WOULD GO HERE)`

likewise, in my websocket that returns progress, I'd hope to read from redis directly

**Comment by vidjuheffex** [comment]
I want to echo this:
on any one job I might want to store:
- ID's (suppose canceling a job entails killing a process... where would I store this)
- progress value
- etc..

Currently, I create entries via `redis` with sane expirys that match `arqs` naming but this feels inelegant.
